src/frequenz/client/electricity_trading/_client.py

```diff
@@ -49,6 +50,32 @@ class _Sentinel:
 NO_VALUE = _Sentinel()
 
 
+def validate_decimal_places(value: Decimal, decimal_places: int, name: str) -> None:
+    """
+    Validate that the decimal places of a given value do not exceed a specified limit.
+
+    Args:
+        value: The value to be checked.
+        decimal_places: The maximum allowed decimal places.
+        name: The name of the value (for error messages).
+
+    Raises:
+        ValueError: If the value has more decimal places than allowed.
+        ValueError: If the value is not a valid decimal number.
+    """
+    assert decimal_places >= 0, "The decimal places must be a non-negative integer."
+
+    if value is not None:
+        try:
+            exponent = int(value.as_tuple().exponent)
+            if abs(exponent) > decimal_places:
+                raise ValueError(
+                    f"The {name} cannot have more than {decimal_places} decimal places."
+                )
+        except InvalidOperation as exc:
+            raise ValueError(f"The {name} is not a valid decimal number.") from exc
```

Comment:
I strongly recommend including the value you are validating here, like: `f"{value} cannot have more than {decimal_places} decimal places"`.